### PR TITLE
BXMSDOC-7455 added section on disabling HTTP with TLS

### PR DESCRIPTION
**Pull request by emmurphy1**
[BXMSDOC-7455](https://issues.redhat.com/browse/BXMSDOC-7455)
[RHPAM draft](http://file.ork.redhat.com/~emmurphy/BXMSDOC-BXMSDOC-7455-pam/#kie-server-smart-router-enable-tls-support-proc_clustering-runtime-standalone)
No impact on RHDM

### CODE REVIEW

File: doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc [hunk 1]
```diff
@@ -14,15 +14,38 @@ xref:clustering-smart-router-install-proc_clustering-runtime-standalone[].
...
       -Dorg.kie.server.router.tls.port = <HTTPS_PORT>
       -jar rhpam-7.9.0-smart-router.jar
 ----
 +
-`org.kie.server.router.tls.port` is a property used to configure the HTTPS port. The default HTTPS port value is `9443`.
+In this example, replace the following variables:
+
+* `<KEYSTORE_PATH>`: The path where the keystore will be stored.
+* `<KEYSTORE_PASSWORD>`: The keystore password.
+* `<AKEYSTORE_ALIAS>`: The password used to access values stored with the alias.
```
Comment (sutaakar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]:
```suggestion
* `<KEYSTORE_ALIAS>`: The alias name used to store the certificate
```

File: doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc [hunk 2]
```diff
@@ -14,15 +14,38 @@ xref:clustering-smart-router-install-proc_clustering-runtime-standalone[].
...
+----
+java  -Dorg.kie.server.router.tls.keystore = <KEYSTORE_PATH>
+      -Dorg.kie.server.router.tls.keystore.password = <KEYSTORE_PASSWORD>
+      -Dorg.kie.server.router.tls.keystore.keyalias = <KEYSTORE_ALIAS>
+      -Dorg.kie.server.router.tls.port = <HTTPS_PORT>
+      -Dorg.kie.server.router.port=0
+      -jar rhpam-7.9.0-smart-router.jar
+----
++
+When the `org.kie.server.router.port` system property is set to `0` or less, then the HTTP listener is not registered. If TLS is configured then Smart Router listens on HTTPS port.
```
Comment (sutaakar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]:
```suggestion
When the `org.kie.server.router.port` system property is set to `0` or less, then the HTTP listener is not registered. If TLS is configured then Smart Router listens only on HTTPS port.
```

File: doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [id='kie-server-smart-router-enable-tls-support-proc_{context}']
 = Configuring Smart Router for TLS support
 
-You can configure Smart Router (KIE Server Router) for TLS support to allow HTTPS traffic.
+You can configure Smart Router (KIE Server Router) for Transport Layer Security (TLS) support to allow HTTPS traffic. In addition, you can disable non-secure HTTP connections to Smart Router.
```
Comment (csherrar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]: [style] Would "unsecured" be better in place of "non-secured"? Maybe rephrase for translatability?

s/In addition, you can disable non-secure HTTP connections to Smart Router/In addition, you can disable unsecured HTTP connections to Smart Router./In addition, you can disable HTTP connections to Smart Router that would not be secure.

Maybe rephrase in terms of the mechanism of Smart Router blocking or rejecting HTTP connections?

"You can also configure Smart Router to reject unsecured HTTP connection attempts."

**Review by sutaakar — APPROVED** [review]

File: doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc [hunk 4]
```diff
@@ -14,15 +14,38 @@ xref:clustering-smart-router-install-proc_clustering-runtime-standalone[].
 endif::[]
 
 .Procedure
-* To start Smart Router with TLS support and HTTPS enabled, use the TLS keystore properties, for example:
+Use one of the following methods to start Smart Router:
```
Comment (csherrar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]: [style] This was brought up during our latest workshop. When handling nonsequential actions, what is the best phrasing? Example I've seen are:

"To do this, perform one of the following tasks:" (The one I thought we use)
"To do this, complete one of the following:" (misses the following subject)
"Use one of the following methods to do this:"

Example from *Handling nonsequential actions* - IBMSG p.88

----
Click *Configure* to do any of the following tasks:
* Choice 1
* Choice 2
----

I thought we use "To do this, perform one of the following tasks:"

File: doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc [hunk 5]
```diff
@@ -14,15 +14,38 @@ xref:clustering-smart-router-install-proc_clustering-runtime-standalone[].
...
 +
-`org.kie.server.router.tls.port` is a property used to configure the HTTPS port. The default HTTPS port value is `9443`.
+In this example, replace the following variables:
+
+* `<KEYSTORE_PATH>`: The path where the keystore will be stored.
+* `<KEYSTORE_PASSWORD>`: The keystore password.
+* `<KEYSTORE_ALIAS>`: The alias name used to store the certificate.
+* `<HTTPS_PORT>`: The HTTPS port. The default HTTPS port is `9443`.
++
+* To start Smart Router with TLS support and HTTPS enabled and with HTTP connections disabled, enter the following command:
```
Comment (csherrar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc:37]: [style] Use of comma here? wdyt?

"To start Smart Router with TLS support and HTTPS enabled, and with HTTP connections disabled, enter the following command:"

or add "where are"

"To start Smart Router with TLS support and HTTPS enabled, where HTTP connections are disabled, enter the following command:"

File: doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc [hunk 6]
```diff
@@ -14,15 +14,38 @@ xref:clustering-smart-router-install-proc_clustering-runtime-standalone[].
...
+----
+java  -Dorg.kie.server.router.tls.keystore = <KEYSTORE_PATH>
+      -Dorg.kie.server.router.tls.keystore.password = <KEYSTORE_PASSWORD>
+      -Dorg.kie.server.router.tls.keystore.keyalias = <KEYSTORE_ALIAS>
+      -Dorg.kie.server.router.tls.port = <HTTPS_PORT>
+      -Dorg.kie.server.router.port=0
+      -jar rhpam-7.9.0-smart-router.jar
+----
++
+When the `org.kie.server.router.port` system property is set to `0` or less, then the HTTP listener is not registered. If TLS is configured then Smart Router listens only on HTTPS port.
```
Comment (csherrar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]: [technical] Regarding  "is set to `0` or less", are negative numbers valid property values and are they handled correctly by defaulting them to 0? If this is not the case, or if it is clearer, I suggest removing "or less". 

s/When the `org.kie.server.router.port` system property is set to `0` or less, then the HTTP listener is not registered./When the `org.kie.server.router.port` system property is set to `0`, then the HTTP listener is not registered.

Comment (csherrar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]: [technical] Regarding "If TLS is configured", why is setting the system property to 0 required if configuring TLS disables HTTP listening?

Comment (csherrar) [inline doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]: [style] Add article for "HTTPS port" and maybe move "only":

s/If TLS is configured then Smart Router listens only on HTTPS port./If TLS is configured then Smart Router only listens on the HTTPS port.

**Review by csherrar — COMMENTED** [review]
Some style and technical questions, otherwise looks good to me.

File: doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc (hunk 6 above)
Comment (sutaakar) [reply doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]: Yes, the negative value is a valid property too.
IMHO it would be ok to reference just `0` for simplicity.

Comment (sutaakar) [reply doc-content/enterprise-only/admin-and-config/kie-server-smart-router-enable-tls-support-proc.adoc]: TLS configuration doesn't disable HTTP listener.
Having both HTTP and TLS active at the same time is a valid scenario.

Maybe be good to adjust the sentence to something like `If TLS is configured and HTTP listener is not registered then Smart Router listens only on HTTPS port.`
